### ISSUE TITLE
max-len overestimates the length of lines that contain tabs

### ISSUE THREAD

**Issue by nre** [eslint/eslint#4661]
* Version 1.10.3.
* Input:
```javascript
/* eslint max-len: [1, 16, 4, {ignoreComments: true}] */
var one		= 1;
var three	= 3;
/*
16-char line:
1234567890123456

The lines contain tabs:
var one\t\t= 1;
var three\t= 3;
*/
```
* Output:
```
test.js
  2:1  warning  Line 2 exceeds the maximum line length of 16  max-len
  3:1  warning  Line 3 exceeds the maximum line length of 16  max-len

? 2 problems (0 errors, 2 warnings)
```
* The lines do not exceed 16 characters. The problem is that the max-len rule assigns a fixed width to each tab *character*, instead of to each tab *stop*.

I have a PR ready to correct this.

**Comment by eslintbot** [comment]
@nre Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Could you explain a bit more about tab stop vs tab character?

**Comment by nre** [comment]
If the tab width is 4 spaces:
```
Tab stops are always 4 spaces wide:
|   |   |   |   |
1234123412341234

Each tab character is 1, 2, 3 or 4 spaces wide:
    <-- tab character is 4 spaces wide.
1   <-- tab character is 3 spaces wide.
12  <-- tab character is 2 spaces wide.
123 <-- tab character is 1 space wide.
```
But max-len will always replace each tab character with 4 spaces (never 3, 2 or 1), which is why it overestimates the length of some lines.

**Comment by ilyavolodin** [comment]
Ahh.. got it, thanks. In that case, I think it's a reasonable fix.

**Comment by nzakas** [comment]
@nre do you want to submit a pull request for this?

**Comment by nre** [comment]
Working on this.
